mycroft/configuration/config.py

```diff
@@ -203,9 +203,18 @@ def load_config_stack(configs=None, cache=False):
         Returns:
             (dict) merged dict of all configuration files
         """
+        # system administrators can define different constraints in how
+        # configurations are loaded
+        system_conf = LocalConf(SYSTEM_CONFIG).get("system") or {}
+        protected_keys = system_conf.get("protected_keys") or {}
+        protected_remote = protected_keys.get("remote") or []
+        protected_user = protected_keys.get("user") or []
```

Comment:
I can see why we'd split this from a "why not have more choice" perspective but it also seems like we're going to effectively merge remote and user configs anyway. At least in the sense of how a user interacts with it.

If that's the case then it seems like we should do this as a single `protected_keys` attribute, otherwise we're encouraging projects to use them separately, and will then need to deprecate that option soon.

 It feels much cleaner to think of:
- Mycroft defaults
- Platform defaults and requirements (as defined by the protected_keys)
- User settings